### DISCUSSION TITLE
在内网使用conan，安装xmake_generator

### DISCUSSION THREAD

**Discussion by penguin-madagascar** [xmake-io/xmake#3847] [Q&A] [answered]
**最终目的为使用xmake在内网通过conan安装package**

在执行如下lua文件后：
```
add_rules("mode.debug", "mode.release")

add_requires("conan::openssl/3.0")

target("test_xrepo")
    set_kind("binary")
    add_files("src/*.cpp")
    add_packages("conan::openssl")
```
出现错误：xmake_generator无法被找到
> ERROR: Unable to find 'xmake_generator/0.1.0@bincrafters/testing' in remotes
> compiler=Visual Studio
> compiler.runtime=MT
> compiler.version=17
> os=Windows
> os_build=Windows
> [options]
> [build_requires]
> [env]
> xmake_generator/0.1.0@bincrafters/testing: Not found in local cache, looking in remotes...
> xmake_generator/0.1.0@bincrafters/testing: Trying with 'repo1'...
> xmake_generator/0.1.0@bincrafters/testing: Trying with 'repo2'...
> if you want to get more verbose errors, please see:
>   -> D:\test_xrepo\build\.packages\c\conan_openssl\3.0\latest\cache\installdir.failed\logs\install.txt
> error: install failed!

目前已将xmake_generator和xmake_installer的github源码下载到本地，但xmake_installer由于无法连接外网会在build处出错：
```
def build(self):
        if self.settings.os_build == "Windows":
            tools.download("https://github.com/tboox/xmake/releases/download/v2.2.3/xmake-v2.2.3.exe", "xmake.exe")
       # else:
            # need simple instructions on build from source with standard build tools
```
正在寻找替代方案，请问有没有更简便的实现方法

提前致谢！

**Comment by waruqi** [comment]
conan 2不会再用这个，内置了 generator

**Reply by penguin-madagascar** [reply]
conan 2的话也会出现相似的问题，在运行提到的lua文件后，出现错误：

> PS D:\test_xrepo> xmake
> checking for Microsoft Visual Studio (x64) version ... 2022
> note: install or modify (m) these packages (pass -y to skip confirm)?
> in conan:
>   -> conan::openssl/3.0 latest [vs_runtime:"MT"]
> please input: y (y/n/m)
> 
>   => install conan::openssl/3.0 latest .. failed
> 
> ======== Input profiles ========
> Profile host:
> [settings]
> arch=x86_64
> build_type=Release
> compiler=msvc
> compiler.runtime=static
> compiler.runtime_type=Release
> compiler.version=193
> os=Windows
> Profile build:
> [settings]
> arch=x86_64
> build_type=Release
> compiler=msvc
> compiler.runtime=static
> compiler.runtime_type=Release
> if you want to get more verbose errors, please see:
>   -> D:\test_xrepo\build\.packages\c\conan_openssl\3.0\latest\cache\installdir.failed\logs\install.txt
> error: install failed!
> 

点开上文提到的路径后，文件显示：

> ERROR: Unable to find 'xmake_generator/0.1.0@bincrafters/testing' in remotes
> 
> compiler=Visual Studio
> compiler.runtime=MT
> compiler.version=17
> os=Windows
> os_build=Windows
> [options]
> [build_requires]
> [env]
> 
> xmake_generator/0.1.0@bincrafters/testing: Not found in local cache, looking in remotes...
> xmake_generator/0.1.0@bincrafters/testing: Trying with 'repo1'...
> xmake_generator/0.1.0@bincrafters/testing: Trying with 'repo2'...

**Reply by penguin-madagascar** [reply]
2.0.6版本conan，2.7.9版本xmake，均为最新版本

**Reply by waruqi** [reply, marked as answer]
还是执行到了 conan1 的逻辑，没检测到 conan2 。。可以尝试 xmake f -c 清下缓存，重新检测下。。不行就自己调下。。https://github.com/xmake-io/xmake/blob/68153d06e61342aa2be927116c9b7d4e40125945/xmake/modules/package/manager/conan/install_package.lua#L39

**Reply by penguin-madagascar** [reply]
非常感谢！！！
